src/Features/Core/Portable/Completion/CompletionService.ProviderManager.cs

```diff
@@ -53,15 +63,34 @@ public ProviderManager(CompletionService service)
                 return _lazyImportedProviders;
             }
 
-            public static ImmutableArray<CompletionProvider> GetProjectCompletionProviders(Project? project)
+            private ValueTask ProcessBatchAsync(ImmutableSegmentedList<IReadOnlyList<AnalyzerReference>> referencesList, CancellationToken cancellationToken)
+            {
+                foreach (var references in referencesList)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    _ = ProjectCompletionProvider.GetExtensions(_service.Language, references);
+                }
+
+                return ValueTaskFactory.CompletedTask;
+            }
+
+            private ImmutableArray<CompletionProvider> GetProjectCompletionProviders(Project? project)
```

Comment:
Not sure if this should have some stronger name like GetProjectCompletionProvidersOrQueueIfNotReady just to make it clear there's a very implicit avoidance of async here.

Reply:
sure!